Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.js

```diff
@@ -67,24 +74,107 @@ gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawStar = function(x1, y1,
+    }
+    this._graphics.endFill();
+};
+*/
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawBezierCurve = function(x1, y1, cpX, cpY, cpX2, cpY2, x2, y2) {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(x1, y1);
+    this._graphics.bezierCurveTo(cpX, cpY, cpX2, cpY2, x2, y2);  
     this._graphics.endFill();
 };
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawQuadraticCurve = function(x1, y1, cpX, cpY, x2, y2) {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(x1, y1);
+    this._graphics.quadraticCurveTo(cpX, cpY, x2, y2);
+    this._graphics.endFill();
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.beginFillPath = function() {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    /*if(!this.hasBeginFillPath){
+        this.beginFillPath();
+        this._graphics.haveBeginFillPath = true;
+    }*/
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.endFillPath = function() {
+   /* if(!this.hasBeginFillPath){
+        this.beginFillPath();
+    }*/
+    this._graphics.endFill();
+    //this._graphics.haveBeginFillPath = false;
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathMoveTo = function(x1, y1) {
+    this._graphics.moveTo(x1, y1);
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathLineTo = function(x1, y1) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.lineTo(x1, y1);
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathBezierCurveTo = function(cpX, cpY, cpX2, cpY2, toX, toY) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.bezierCurveTo(cpX, cpY, cpX2, cpY2, toX, toY);  
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathArc = function(x1, y1, radius, startAngle, endAngle, anticlockwise) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.arc(x1, y1, radius, gdjs.toRad(startAngle), gdjs.toRad(endAngle), anticlockwise ? true : false);
+};
+/*
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathArcTo = function(x1, y1, x2, y2, radius) {
+    this._graphics.arcTo(x1, y1, x2, y2, radius);
+};
 */
 
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathQuadraticCurveTo = function(cpX, cpY, toX, toY) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.quadraticCurveTo(cpX, cpY, toX, toY);  
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.closePath = function() {
+    this._graphics.closePath();  
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.hasBeginFillPath = function() {
+    if(this._graphics.haveBeginFillPath){
```

Comment:
`haveBeginFillPath` should be renamed to `hasBegunFillPath`

Reply:
Also, you can just `return !!this._graphics.hasBegiuFillPath;` (the `!!` to make sure that if it's `undefined`, it will be "converted" to false).